File: Services/Student/Student.Infrastructure/Extensions/DapperConnectionProvider.cs
using Dapper;
using Npgsql;
using Student.Application.Settings;

namespace Student.Infrastructure.Extensions;

public static class DapperConnectionProvider
{
    public static async Task<NpgsqlConnection> ConnectionAsync()
    {
        SimpleCRUD.SetDialect(SimpleCRUD.Dialect.PostgreSQL);
        return await Task.FromResult(new NpgsqlConnection(DatabaseSettings.ConnectionString));
    }
}

File: Services/Teacher/Teacher.Infrastructure/Extensions/DapperConnectionProvider.vb
Imports Dapper
Imports Npgsql
Imports Teacher.Application
Public Module DapperConnectionProvider
    Public Async Function ConnectionAsync() As Task(Of NpgsqlConnection)
        SimpleCRUD.SetDialect(SimpleCRUD.Dialect.PostgreSQL)
        Return Await Task.FromResult(New NpgsqlConnection(DatabaseSettings.ConnectionString))
    End Function
End Module
